lib/xgu/xgux.h

```diff
@@ -0,0 +1,88 @@
+#ifndef __XGUX_H
+#define __XGUX_H
+
+#include "xgu.h"
+
+inline
+void xgux_draw_arrays(XguPrimitiveType mode, unsigned int start, unsigned int count) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+    p = xgu_draw_arrays(p, start, count);
+    p = xgu_end(p);
+    pb_end(p);
+}
+
+inline
+void xgux_draw_elements16(XguPrimitiveType mode, unsigned int count, const uint16_t* elements) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+
+    /* Submit elements in pairs if possible */
+    unsigned int pair_count = count / 2;
+    for (unsigned int i = 0; i < pair_count; ) {
+
+        if (i > 0) {
+            /* Start next batch */
+            p = pb_end(p);
+            p = pb_begin();
+        }
+
+        unsigned int batch_pair_count = MIN(MAX_BATCH, pair_count - i);
```

Comment:
`MAX_BATCH` and `MIN` macro are missing?

They were here in #7:

https://github.com/dracc/nxdk/blob/bbcf70648e395b3b2c61329e1adab00c4cc68a36/lib/xgu/xgux.h#L5-L8